waveanalysis/data_workflows/rolling_workflow.py

```diff
@@ -0,0 +1,262 @@
+                                shift = sp.calc_indv_shift(cc_curve=ccf)
+                                average_period = np.mean(indv_periods[:, :, bin])
+                                shift = sp.small_shifts_correction(delay_frames=shift, average_period=average_period)
+                                indv_shifts[submovie, combo_number, bin] = shift
+
+            # create a subfolder within the main save path with the same name as the image file
+            im_save_path = os.path.join(main_save_path, name_wo_ext)
+            os.makedirs(im_save_path, exist_ok=True)
+
+            # adjust the different waves properties to be the use the frame interval rather than the number of frames
+            indv_periods = indv_periods * img_props_dict['frame_interval']
+            indv_peak_offsets = indv_peak_offsets * img_props_dict['frame_interval']
+            indv_peak_widths = indv_peak_widths * img_props_dict['frame_interval']
+
+            img_parameters_dict = {
+                            'Period': indv_periods,
+                            'Peak Amp': indv_peak_amps,
+                            'Peak Rel Amp': indv_peak_rel_amps,
+                            'Peak Width': indv_peak_widths,
+                            'Peak Max': indv_peak_maxs,
+                            'Peak Min': indv_peak_mins,
+                            'Peak Offset': indv_peak_offsets
+            }
+
+            # add shifts to the dictionary if there are multiple channels
+            if img_props_dict['num_channels'] > 1:
+                indv_shifts = indv_shifts * img_props_dict['frame_interval']
+                img_parameters_dict['Shift'] = indv_shifts
+                img_parameters_dict['% Phase Shift'] = indv_shifts / indv_periods
+
+            # calculate the number of subframes used
+            log_params['Submovies Used'].append(num_submovies)
+
+            ############################################
+            ############## Saving and Summary ##########
+            ############################################
+
+            # summarize the data for each subframe as individual dataframes, and save as .csv
+            submovie_meas_list, _ = summarize_image(
+                img_props_dict=img_props_dict,
+                img_parameters=img_parameters_dict
+            )
+            csv_save_path = os.path.join(im_save_path, 'rolling_measurements')
+            os.makedirs(csv_save_path, exist_ok=True)
+            for measurement_index, submovie_meas_df in enumerate(submovie_meas_list):  # type: ignore
+                submovie_meas_df.to_csv(f'{csv_save_path}/{name_wo_ext}_subframe{measurement_index}_measurements.csv', index = False)
+            
+            # summarize the data for each subframe as a single dataframe, and save as .csv
+            summary_df = combine_stats_rolling(
+                img_props_dict=img_props_dict,
+                img_parameters_dict=img_parameters_dict,
+                indv_ccfs=indv_ccfs
```

Comment:
`indv_ccfs` will be unbound if the dataset is only 1 channel. I ran into this because the sample dataset in `assets/data/test_data_rolling/` is 1 channel so you can use that for testing. You can probably make the argument to this function optional and then either pass `indv_ccfs` or None at runtime depending on the dataset `indv_ccfs=indv_ccfs if num_channels > 1 else None`